src/evict/evict_page.c

```diff
@@ -73,47 +67,20 @@ __wt_page_release_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 * without first locking the page, it could be evicted in between.
 	 */
 	previous_state = ref->state;
-	if ((previous_state == WT_REF_MEM || previous_state == WT_REF_LIMBO) &&
-	    WT_REF_CAS_STATE(session, ref, previous_state, WT_REF_LOCKED))
-		locked = true;
+	locked =
+	    (previous_state == WT_REF_MEM || previous_state == WT_REF_LIMBO) &&
+	    WT_REF_CAS_STATE(session, ref, previous_state, WT_REF_LOCKED);
 	if ((ret = __wt_hazard_clear(session, ref)) != 0 || !locked) {
 		if (locked)
 			WT_REF_SET_STATE(ref, previous_state);
 		return (ret == 0 ? EBUSY : ret);
 	}
 
-	(void)__wt_atomic_addv32(&btree->evict_busy, 1);
-
-	too_big = page->memory_footprint >= btree->splitmempage;
+	evict_flags = LF_ISSET(WT_READ_NO_SPLIT) ? WT_EVICT_CALL_NO_SPLIT : 0;
+	FLD_SET(evict_flags, WT_EVICT_CALL_URGENT);
 
-	/*
-	 * Track how long the call to evict took. If eviction is successful then
-	 * we have one of two pairs of stats to increment.
-	 */
+	(void)__wt_atomic_addv32(&btree->evict_busy, 1);
 	ret = __wt_evict(session, ref, previous_state, evict_flags);
-	time_stop = __wt_clock(session);
-	if (ret == 0) {
-		if (too_big) {
```

Comment:
Did you have to push this down a level to know reliably whether the page was clean or dirty?

Reply:
We can figure it out at a higher level (by calling `__wt_page_evict_clean()`), but that function is a kluge, not excited about using it in more places.

More importantly, all the other statistics live in `__wt_evict()` and there's no advantage in in-lining that code in a bunch of places.

In summary, it just seemed cleaner to handle all the statistics together in the same place.